### PR TITLE
Fix scalar type of sum

### LINKED ISSUE #664: reduction on preds
```
import torch
import torch_xla
import torch_xla_py.utils as xu
import torch_xla_py.xla_model as xm
import pdb
xla_device = xm.xla_device()
# xla_device = 'cpu'
a = torch.rand(61440).to(xla_device) - 0.5
pdb.set_trace()
print((a < 0).sum())
print((a == 0).sum())
print((a > 0).sum())
```

### PR DESCRIPTION
**Pull request by asuhan**
Fixes #664.

### CODE REVIEW

File: torch_xla/csrc/tensor_methods.cpp [hunk 1]
```diff
@@ -1889,11 +1889,15 @@ XLATensor XLATensor::sum(const XLATensor& input,
                          std::vector<xla::int64> dimensions,
                          bool keep_reduced_dimensions,
                          c10::optional<at::ScalarType> dtype) {
+  if (at::isIntegralType(input.dtype()) && !dtype) {
+    dtype = at::ScalarType::Long;
```
Comment (dlibenzi) [inline torch_xla/csrc/tensor_methods.cpp]: Hmm, why not letting it get input.dtype instead of bolting in kLong?

Comment (asuhan) [reply torch_xla/csrc/tensor_methods.cpp]: Because that the PyTorch semantics. If the input type is any sort of integral, the output type is a long.

Comment (dlibenzi) [reply torch_xla/csrc/tensor_methods.cpp]: So int+int == long?

Comment (asuhan) [reply torch_xla/csrc/tensor_methods.cpp]: Yes.

**Pull request merged**